src/include/duckdb/common/serializer/serialization_traits.hpp

```diff
@@ -124,6 +124,12 @@ struct is_optional_ptr<optional_ptr<T>> : std::true_type {
 	typedef T ELEMENT_TYPE;
 };
 
+template <typename T>
+struct is_pointer : std::false_type {};
+
+template <typename T>
+struct is_pointer<T *> : std::true_type {};
```

Comment:
Can this not be either an `optional_ptr`, or a reference?

Reply:
I can turn it into a reference 👍 
If I turn it into an `optional_ptr` I am just shifting the problem, because I can already create an unowned unique_ptr on the spot (which is medium hacky), but it takes a `const &`, not a `&&` so that didn't work

Reply:
Though if we accept a reference, then we can't serialize a null anymore, which is not the intention here